src/config.rs

```diff
@@ -662,6 +669,38 @@ impl Default for DbOptions {
     }
 }
 
+#[derive(Clone, Deserialize, Serialize)]
+pub struct DbReaderOptions {
+    /// How frequently to poll for new manifest files. Refreshing the manifest file allows readers
+    /// to detect newly compacted data.
+    pub manifest_poll_interval: Duration,
+
+    /// For readers that refresh their checkpoint, this specifies the lifetime to use for the
+    /// created checkpoint. The checkpoint’s expire time will be set to the current time plus
+    /// this value. If not specified, then the checkpoint will be created with no expiry, and
+    /// must be manually removed. This lifetime must always be greater than
+    /// manifest_poll_interval x 2
+    pub checkpoint_lifetime: Option<Duration>,
```

Comment:
I am considering making this required for readers. I am not sure how a user would go about manually removing a checkpoint that was created by the reader. Search through the logs for the ID perhaps? It seems nice to know that cleanup is eventually guaranteed. I guess users can still choose a very long lifetime if they really want that behavior.

Reply:
I went ahead and did this since it gives us a nice way to provide a reasonable default. This gives sensible behavior if the user just provides `DbReaderOptions::default()`.